src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -106,15 +106,15 @@ public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
       } else {
         deleteFile(nameDir);
         nameDir.mkdirs();
-        runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
-        startProcess(driver, nameNodeTask);
-        startProcess(driver, zkfcNodeTask);
-        driver.sendStatusUpdate(TaskStatus.newBuilder()
-            .setTaskId(nameNodeTask.taskInfo.getTaskId())
-            .setState(TaskState.TASK_RUNNING)
-            .setMessage(messageStr)
-            .build());
       }
+      runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
+      startProcess(driver, nameNodeTask);
+      startProcess(driver, zkfcNodeTask);
+      driver.sendStatusUpdate(TaskStatus.newBuilder()
+          .setTaskId(nameNodeTask.taskInfo.getTaskId())
+          .setState(TaskState.TASK_RUNNING)
+          .setMessage(messageStr)
+          .build());
```

Comment:
So now the "hdfs-mesos-namenode -i/b" is only in the else case. That's the part that does the reformatting, right?
Seems like if the scheduler sends a frameworkMessage(init/bootstrap) to the node, then it should send back the TASK_RUNNING(init/bootstrap) regardless of whether it reformatted. But real data could convince me otherwise.
The question of whether/when you need to startProcess(NN,ZKFC) is still uncertain to me.